pkg/targetgroupbinding/targets_manager.go

```diff
@@ -199,6 +206,16 @@ func (m *cachedTargetsManager) listTargetsFromAWS(ctx context.Context, tgARN str
 
 	listedTargets := make([]TargetInfo, 0, len(resp.TargetHealthDescriptions))
```

Comment:
the original assumption in this make is that capacity of slice will be == to the len of the targets, but that's no longer true if we skip any - will this broken assumption cause any other problems? Should we no longer make this way because the assumption doesn't hold?

Reply:
we make a fixed length array then append at the end...

Reply:
yea